### DISCUSSION TITLE
Running localy on windows

### DISCUSSION THREAD

**Discussion by BurmeseCat** [cotes2020/jekyll-theme-chirpy#1307] [Q&A] [answered]
It is possible to run this locally on windows?

**Comment by DmitriyFrogo** [comment, marked as answer]
Yes

Read this post:
https://chirpy.cotes.page/posts/getting-started/
Especially this part:
[#running-local-server](https://chirpy.cotes.page/posts/getting-started/#running-local-server)

Also, if you want to host it through LAN or Internet, add these lines to the end of `_config.yml` and change it to your IP and desired port:

```
port                : 4000
host                : 127.0.0.1
```

Also you need to forward desired port in your router control panel.
Otherwise, you won't be able to open this site from other devices.

**Comment by DmitriyFrogo** [comment]
TL;DR

1. Install latest [Ruby+Devkit](https://rubyinstaller.org/downloads/)
- Just click Next and wait, until console appears
- In console type 3 and press Enter (Option 3 means "MSYS2 and MINGW development toolchain")
- Wait, press Enter

2. Install latest [Git](https://git-scm.com/download/win)
- Next, Next, Next...

3. [chirpy-starter](https://github.com/cotes2020/chirpy-starter) -> Code -> [Download ZIP](https://github.com/cotes2020/chirpy-starter/archive/refs/heads/main.zip)
- Unpack somewhere

4. Open folder with chirpy-starter
- If you on Win 10/11, just right click on empty space - Open in Terminal

5. Type `bundle`, press Enter, and wait, until it downloading all dependencies

6. Type `bundle exec jekyll serve`
- Now you can connect to localhost:4000 and make real-time changes in your files

7. If you change your _config.yml, you also need to know how to restart server
- `CTRL+C`, then type `y`, then Enter to stop your server
- Then type (or just press up arrow button on your keyboard) `bundle exec jekyll serve` to start it again

**Reply by DmitriyFrogo** [reply]
If you already have repo, just clone it with Git or Github Desktop, and skip step 3

**Reply by BurmeseCat** [reply]
It works. Thanks you so much!
